micro-rdk/src/common/data_manager.rs

```diff
@@ -658,41 +712,66 @@ impl DataCollectAndSyncTask {
                         err
                     )
                 });
-            if let Ok(mut readings) = readings {
-                match time_correct_reading(self.robot_start_time, &mut readings) {
-                    Ok(_) => {
-                        let upload_request = DataCaptureUploadRequest {
-                            metadata: Some(UploadMetadata {
-                                part_id: self.part_id.clone(),
-                                component_type: collector_key.component_type.clone(),
-                                r#type: DataType::TabularSensor.into(),
-                                component_name: collector_key.r_name.clone(),
-                                method_name: collector_key.method.to_string(),
-                                ..Default::default()
-                            }),
-                            sensor_contents: vec![readings],
-                        };
-                        match app_client.upload_data(upload_request).await {
-                            Ok(_) => {
-                                #[cfg(feature = "data-upload-hook-unstable")]
-                                unsafe {
-                                    micro_rdk_data_manager_post_upload_hook();
+            if let Ok(readings) = readings {
+                let robot_start_time = self.robot_start_time;
+                let mut readings: Vec<SensorData> = readings
+                    .into_iter()
+                    .filter_map(|mut readings| {
+                        if let Err(err) = time_correct_reading(robot_start_time, &mut readings) {
+                            log::error!(
+                                "unable to time correct reading for collector {:?}, {:?}",
+                                collector_key,
+                                err
+                            );
+                            None
+                        } else {
+                            Some(readings)
+                        }
+                    })
+                    .collect();
+                readings.reverse();
```

Comment:
why is this needed?

Reply:
just makes it easier to make sure we're uploading readings chronologically. By reversing first, I can just call `pop` later

Reply:
I see. Not a blocker then but if we don't need to send data ordered by time then we should not do it. Add a comment that this is not a requirement if you don't want to change the logic